arangod/Transaction/Helpers.cpp

```diff
@@ -385,18 +385,18 @@ OperationResult transaction::helpers::buildCountResult(std::vector<std::pair<std
     }
     resultBuilder.close();
   }
-  return OperationResult(Result(), resultBuilder.steal(), nullptr, false);
+  return OperationResult(Result(), resultBuilder.buffer(), nullptr);
```

Comment:
is this change intentional?

Reply:
yes

Reply:
why is it needed?

Reply:
This is not necessary because OperationResult takes a `const ref` that prologs the lifetime of the temporary stolen object. The `buffer()` on the other hand does not return the object itself but a reference making it more obvious what is going on and will even work if the constructor's argument should loose it's constness.

Reply:
Further more `steal()` copies the shared_ptr for nothing.